File: 04-herencia/juego_de_estrategia/unidad.py
"""Clase Unidad"""

from __future__ import annotations

from abc import ABCMeta, abstractmethod


class Unidad(metaclass=ABCMeta):
    """Clase abstracta unidad"""

    def __init__(self, salud, danio, posicion):
        self.__salud = salud
        self.__danio = danio
        self.__posicion = posicion

    @property
    def posicion(self):
        return self.__posicion

    @property
    def danio(self):
        return self.__danio

    @property
    def salud(self):
        return self.__salud

    @abstractmethod
    def atacar(self, enemigo: Unidad):
        pass

    def recibir_danio(self, danio):
        self.__salud -= danio

    def esta_muerto(self) -> bool:
        return self.__salud <= 0

    def distancia(self, enemigo: Unidad):
        return abs(self.__posicion - enemigo.posicion)

File: workspace-2025-1c/Herencia/src/juegoDeEstrategia/Unidad.java
package juegoDeEstrategia;

public abstract class Unidad {

	private int salud;
	private int danio;
	private int posicion;

	public int getSalud() {
		return salud;
	}

	public int getDanio() {
		return danio;
	}
	public int getPosicion() {
		return this.posicion;
	}

	public Unidad(int salud, int danio, int posicion) {
		this.salud = salud;
		this.danio = danio;
		this.posicion = posicion;
	}

	

	public void recibirDanio(int danio) {
		this.salud -= danio;
	}

	public boolean estaMuerto() {
		return this.salud <= 0;
	}

	public int distancia(Unidad enemigo) {
		return Math.abs(this.getPosicion() - enemigo.getPosicion());
	}

	public abstract void atacar(Unidad enemigo);

}
